### DISCUSSION TITLE
How to programmatically detect/force console host?

### DISCUSSION THREAD

**Discussion by dts350z** [microsoft/terminal#14698] [Q&A]
So I have a console application that uses SetConsoleWindowInfo, etc. which does nothing in windows terminal.

Windows Terminal is now the default in windows 11, and while users can set ALL console hosts to be windows terminals or the old console, I don't see anyway for console applications to determine which is selected or **to force the right choice for the API in use**. 

"Let Windows Decide" doesn't seem to do anything except make everything windows terminals.

I also don't see a 1:1 equivalency between the old API and Virtual Terminal Escape sequences. E.g. one can set the width, but not the height?

Please advise.

**Comment by zadjii-msft** [comment]
What are you trying to achieve with `SetConsoleWindowInfo`/? I'd personally rather try to find the right way to replicate that functionality in the Terminal, rather than provide an escape hatch. If we really can't find a sensible way of replicating that in the Terminal, well, we can talk about escape hatches then 😉

**Comment by dts350z** [comment]
I use Get/SetConsoleScreenBufferSize to set the buffer

I use SetConsoleWindowInfo to set the window size to  {0, 0, 150	, 20}

I use CONSOLE_SCREEN_BUFFER_INFO with  GetStdHandle(STD_OUTPUT_HANDLE to get the X and Y current cursor position

I use SetConsoleCursorPosition to set the cursor position

I use SetConsoleTextAttribute to set font colors (some works in windows terminal, some does not. Some font colors not visible.)

What I get with Console Host:
![image](https://user-images.githubusercontent.com/5871786/212746767-2c55f92f-8cca-4b0b-8c7a-cea314230543.png)

What I get with Windows Terminal:
![image](https://user-images.githubusercontent.com/5871786/212747245-935267d9-f957-447b-9b3a-39a054c0f559.png)

**Reply by zadjii-msft** [reply]
Ah, okay, there's a bunch of things here:
* "I want to resize the Terminal": That's tracked in #5094. That we're intending to get to.
* "I get and restore the cursor position": That _should already work fine in the Terminal_. If you're seeing issues with this, it might be due to the disconnect between the backing buffer's size and the Terminal's size. 
* "I'm settings text attributes": Again, that should work fine? The only thing I saw was a difference in the red text output. That could just come down to a difference in the color schemes in Terminal and conhost.

I'd mainly expect your issues to come down to #5049. I'd rather fix that than give an escape hatch, honestly.

**Comment by 7thFox** [comment]
Or let's just mention the fact that I pin a frequently used console app to my taksbar and it doesn't stay grouped there when it's launched (instead is under WT) and wouldn't display my icon.
